src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -745,10 +753,51 @@ private static CipherSuite chooseCipherSuite(
             }
 
             // no cipher suites in common
+            if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
+                printServerSocketConfig(shc, null);
+            }
             return null;
         }
     }
 
+    // print out server socket config
+    private static void printServerSocketConfig(ServerHandshakeContext shc,
+                                                       List<CipherSuite> legacySuites) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("\"{0}\": '\n{'\n")
+                .append("  \"preferred ciphersuites\"     : \"{1}\",\n")
+                .append("  \"client auth type\"           : \"{2}\",\n")
+                .append("  \"enabled server ciphersuites\": \"{3}\",\n")
+                .append("  \"legacy algorithms\"          : \"{4}\"");
+
+        LinkedList<String> fieldsList = new LinkedList<>();
+
+        fieldsList.add("SSLSeverSocket info");
+        fieldsList.add(shc.sslConfig.preferLocalCipherSuites ? "Server preference" : "Client preference");
+        fieldsList.add(shc.sslConfig.clientAuthType.toString());
+        fieldsList.add(shc.activeCipherSuites != null ? shc.activeCipherSuites.toString() : "Not Set");
+        fieldsList.add(Security.getProperty(LegacyAlgorithmConstraints.PROPERTY_TLS_LEGACY_ALGS));
+
+        if (!shc.negotiatedProtocol.name.equalsIgnoreCase(ProtocolVersion.TLS13.name)) {
```

Comment:
This check is somewhat incorrect in my opinion, just because none of the legacy algs don't apply to TLS 1.3 doesn't mean that we won't add one in the future or someone could configure it themselves. It feels like this is the wrong place to be logging about legacy/disabled suites and this should be logged only if the constraints have been applied. I would remove the logging of the legacy algs as it doesn't seem critical. Plus it doesn't make sense to only log legacy algs and not disabled algs.

Reply:
Looks like the TLS Legacy algs and the disabled cipher suites values are final static. I think we should remove them from this patch and open a new JBS enhancement. The new issue can log these two final values when set by the JDK -- useful values to know for debug logs.